### PR TITLE
BAU - Code Hygiene/Style - Add spotless plugin

### PR DESCRIPTION
**Pull request by tal-nagra**
## Proposed changes
Introduce `spotless` to ensure code has the same format/style irrespective of who it is authored by.
### What changed
The spotless plugin has been added to build.gradle 
More info here:
https://github.com/diffplug/spotless/tree/main/plugin-gradle#quickstart

### Why did it change
To ensure a consistent code style.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -59,9 +67,10 @@ sonarqube {
...
-        xml.enabled true
+spotless {
+    java {
+        target "src/**/*.java"
+        googleJavaFormat("1.13.0").aosp()
+        importOrder "", "javax", "java", "\\#"
     }
-}
-test.finalizedBy jacocoTestReport
+}
```
Comment (dannyhayes-gds) [inline build.gradle]: Needs a EOF line here

**Review by jamietanna — COMMENTED** [review]
Is this replacing #42?

File: build.gradle [hunk 2]
```diff
@@ -59,9 +67,10 @@ sonarqube {
...
 }
 
-jacocoTestReport {
-    reports {
-        xml.enabled true
+spotless {
+    java {
+        target "src/**/*.java"
+        googleJavaFormat("1.13.0").aosp()
+        importOrder "", "javax", "java", "\\#"
```
Comment (jamietanna) [inline build.gradle]: Let's enforce the newline requirement:
```suggestion
        importOrder "", "javax", "java", "\\#"
        endWithNewline()        
```

File: build.gradle [hunk 3]
```diff
@@ -59,9 +67,10 @@ sonarqube {
     }
 }
 
-jacocoTestReport {
-    reports {
-        xml.enabled true
+spotless {
```
Comment (jamietanna) [inline build.gradle:72]: should we also add `groovyGradle` formatting?

File: build.gradle [hunk 4]
```diff
@@ -1,67 +1,80 @@
 plugins {
-    id "java"
-    id "application"
-    id "idea"
-    id "org.sonarqube" version "3.3"
-    id "jacoco"
+	id "java"
+	id "application"
+	id "idea"
```
Comment (jamietanna) [inline build.gradle:4]: As an aside, I'd recommend folks add this in their own Gradle config i.e. https://gitlab.com/jamietanna/dotfiles-arch/-/blob/master/java/home/.gradle/init.d/idea.gradle so we don't need to add unnecessary plugins for IDEs (especially as not everyone will use it!)

File: build.gradle [hunk 5]
```diff
@@ -1,67 +1,80 @@
...
+spotless {
+	java {
+		target "src/**/*.java"
+		googleJavaFormat("1.13.0").aosp()
+		importOrder "", "javax", "java", "\\#"
+		endWithNewline()
+	}
+	groovyGradle {
+		target '*.gradle' // default target of groovyGradle
+	}
```
Comment (jamietanna) [inline build.gradle]: I would recommend at least:

```suggestion
  groovyGradle {
    target '**/*.gradle'
    trimTrailingWhitespace()
    endWithNewline()
  }
```

But probably:

```suggestion
  groovyGradle {
    target '**/*.gradle'
    greclipse()
    trimTrailingWhitespace()
    endWithNewline()
  }
```

Comment (tal-nagra) [reply build.gradle]: Amended to add groovyGradle

File: build.gradle (hunk 3 above)
Comment (tal-nagra) [reply build.gradle:72]: groovyGradle now added

File: build.gradle (hunk 2 above)
Comment (tal-nagra) [reply build.gradle]: added `endWithNewLine()`

File: build.gradle (hunk 1 above)
Comment (tal-nagra) [reply build.gradle]: Now corrected

**Comment by tal-nagra** [comment]
> Is this replacing #42?

Yes

**Review by jamietanna — APPROVED** [review]
Looks good- needs someone with commit access to approve :+1:

**Review by kerrr — APPROVED** [review]

**Pull request merged**